### ISSUE TITLE
Type error when sending `Readable` to `reply.send`

### ISSUE THREAD

**Issue by theguacamoleking** [fastify/fastify-type-provider-json-schema-to-ts#108]
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the bug has not already been reported

### Fastify version

5

### Plugin version

4.0.1

### Node.js version

20

### Operating system

Linux

### Operating system version (i.e. 20.04, 11.3, 10)

n/a

### Description

The following schema

```js
{
  schema: {
    // ...
    response: {
      200: {
        type: 'string',
        headers: {
          'Content-Type': { type: 'application/octet-stream', default: 'text/csv' },
        },
      },
    },
  },
}
```

Leads to `reply.send` being typed as `(payload: string | undefined) => ...`.

This is problematic because sending [streams](https://fastify.dev/docs/v1.14.x/Documentation/Reply/#streams) leads to a type error

> Argument of type `Readable` is not assignable to parameter of type `string`.

### Expected Behavior

`reply.send(someReadable)` does not lead to a type error

**Comment by bcomnes** [comment]
How would you expect to represent this with [json-schema-to-ts](https://github.com/ThomasAribart/json-schema-to-ts)?

**Comment by Tony133** [comment]
Closing as not planned. A JSON Schema with `type: 'string'` correctly infers `string`, and there's no way to express a `Readable` in JSON Schema, so this isn't something the type provider can infer. For stream responses, either omit the `response` schema for that route or cast (`reply.send(stream as any)`).

The underlying limitation is in Fastify core's `SendArgs`, and is now tracked in fastify/fastify#7009 (proposed fix in fastify/fastify#7010). Once that lands, streams will work with any type provider without changes here.

If either workaround still fails for you, please open a new issue with a minimal reproduction in a cloneable git repository.
